### ISSUE TITLE
App within Solution Checksum Fail on Unpack

### ISSUE THREAD

**Issue by AlexBrownShi** [microsoft/PowerApps-Language-Tooling#355]
We have a solution that contains an App.  We unpack the solution, then we have to unpack the app.  
When we unpack the app we receive:

UnPack from E:\Repos\PartnerCertifications\PartnerCertifications\CanvasApps\new_pcap_7bb8d_DocumentUri.msapp-->E:\Repos\PartnerCertifications\CanvasUnpacked\new_pcap_7bb8d_DocumentUri.msapp
  source format version: 0.24
Error   PA3011: Roundtrip validation on unpack failed. This is always a bug, please file an issue at https://github.com/microsoft/PowerApps-Language-Tooling
Error   PA3011: Roundtrip validation on unpack failed. This is always a bug, please file an issue at https://github.com/microsoft/PowerApps-Language-Tooling
Microsoft PowerPlatform CLI
Version: 1.12.2+gf0f06de

Error: There were errors.

Usage: pac canvas unpack [--sources] --msapp

  --sources                   Directory to sources to be unpacked
  --msapp                     Path to .msapp file

**Comment by satishkotaru** [comment]
We are also facing the same issue even in local and through Pipeline's as well! Any update's? @microsoftopensource @NavneetThekkumpat

**Comment by NavneetThekkumpat** [comment]
@AlexBrownShi / @satishkotaru can you mail me your msapp as a repo, at nthekkumpat at microsoft dot com?

**Comment by mikefactorial** [comment]
We've started seeing this in the CoE Starter Kit ALM Accelerator App as well when unpacking the ALM Accelerator App itself. Will forward the msapp for repro. If it helps, we were able to unpack successfully using PAC CLI as recently as February 19th which was the last time we tried since today.

**Comment by PatrickMichaelKriegler** [comment]
We are also facing this issue. We previously got the same generic Roundtrip error when we were using a Teams component in one of our canvasapps. We removed that component and everything worked well for a few weeks. We haven't added any Teams or custom components that we suspect are provoking it. The one thing that has been added to the apps that may be a bit out of the ordinary is ModelDrivenFormIntegration.

We have a total of 5 canvas apps in our solution. When unpacking the individual msapp files locally with PAC CLI  3 of the 5 fail with the Round trip error.  The other 2 also fail but with a different error:

'Sorry, the app encountered a non recoverable error and will need to terminate. The exception details have been captured and will be forwarded to the development team, if telemetry has been enabled. Exception Id: 83984489-42f6-419a-bb41-b430c1c7e05f, Exception Type: System.IO.DirectoryNotFoundException'

So far we havn't figured out what could be the common denominator for the 3 that have the Roundtrip Validation error. Not all of them use the ModelDrivenFormIntegration and 2 of the 3 are new but very small and non complex apps.
